platform/lang-impl/src/com/intellij/ide/GeneratedFileEditingNotificationProvider.java

```diff
@@ -16,33 +15,25 @@
 import javax.swing.*;
 import java.util.function.Function;
 
+import static com.google.common.base.MoreObjects.firstNonNull;
+
 final class GeneratedFileEditingNotificationProvider implements EditorNotificationProvider, DumbAware {
   @Override
   public @Nullable Function<? super @NotNull FileEditor, ? extends @Nullable JComponent> collectNotificationData(@NotNull Project project,
                                                                                                                  @NotNull VirtualFile file) {
-    if (!GeneratedSourceFileChangeTracker.getInstance(project).isEditedGeneratedFile(file)) return null;
+    var matchingFilter = GeneratedSourcesFilter.findFirstMatchingFilterNonBlocking(file, project);
+
+    if (matchingFilter == null) return null;
 
-    String notificationText = getText(file, project);
+    String notificationText = firstNonNull(
```

Comment:
Would be nice to use platform's util method `com.intellij.util.ObjectUtils.notNull`

Reply:
I assume this is because of the `Contract`s for static verification?

Reply:
I just think that in the context of developing the platform, it is slightly easier to understand what intellij's `notNull` does than google's `firstNonNull`